hacks.js

```diff
@@ -0,0 +1,27 @@
+// Addresses an issue where RTC peers are not properly garbage collected in Chromium:
+// https://bugs.chromium.org/p/chromium/issues/detail?id=825576
+// This error leads to WebTorrent Desktop and Chromium-based browsers being completely
+// unable to seed until the process is restarted:
+// https://github.com/webtorrent/webtorrent/issues/1981
+let gcInterval = 1
+function forcePeerGC () {
+  let peer = new RTCPeerConnection()
+  setTimeout(() => {
+    peer.close()
+    peer = null
+  }, 10)
+  console.log(gcInterval++)
+  if (!(gcInterval % 20)) {
+    queueMicrotask(() => {
+      console.log('Collecting garbage.')
+      let img = document.createElement('img')
+      img.src = window.URL.createObjectURL(new Blob([new ArrayBuffer(5e+7)]))
+      img.onerror = function () {
+        window.URL.revokeObjectURL(this.src)
+        img = null
+      }
+    })
+  }
+}
+
+setInterval(forcePeerGC, 5000)
```

Comment:
Is it necessary to do this interval every 5s or can we instead catch this "too many connection" error and then run the GC?

Reply:
The error doesn't present until LONG after torrents have stopped working. In most cases, you don't even see an error in the console at all. So, there's nothing to catch. But in every case, applying this fix addresses the problem immediately.

Reply:
> Is it necessary to do this interval every 5s or can we instead catch this "too many connection" error and then run the GC?

I think I agree that this runs when it shouldn't, it should check ~30 seconds if there have been any closed connections since the last time this was run, then force GC, all the incrementation stuff and peer creation is useless

Reply:
I agree. I just removed the unnecessary bits. It's still failing tests, but with new errors now.